### PR TITLE
doc: updated 54h documentation

### PR DESCRIPTION
**Pull request by FrancescoSer**
Updated 54h documentation.

### CODE REVIEW

File: doc/nrf/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_matter_thread.rst [hunk 1]
```diff
@@ -137,7 +127,7 @@ Additional requirements on the nRF54H20 PDK
 In addition to the standard requirements for the |NCS|, such as the :ref:`ug_matter_gs_tools_gn` for Matter, you need the following to run Matter-enabled or Thread-enabled applications on the nRF54H20 PDK:
 
 * For DFU - J-Link and a USB cable.
-* The compatible version of the nrfjprog tool, included in the :ref:`nRF Command Line Tools version specific to the limited customer sampling<nrf54h20_install_commandline>`.
+* nrfjprog from the nRF CLI tools.
```
Comment (FrancescoSer) [inline doc/nrf/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_matter_thread.rst:130]: CLI tools need link

**Review by greg-fer — APPROVED** [review]
Step in the right direction. I'll have a closer look at PRs where :orphan: tags are being gradually removed.

File: doc/nrf/glossary.rst [hunk 2]
```diff
@@ -145,13 +155,20 @@ Glossary
    Contribution
       A change to the codebase sent to a remote :term:`repository <Repository>` for inclusion.
 
+   Core
+      Subsets of domains.
```
Comment (wiba-nordic) [inline doc/nrf/glossary.rst:159]:
```suggestion
      Subsets of :term:`domains <Domain>`.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/glossary.rst [hunk 3]
```diff
@@ -663,6 +743,19 @@ Glossary
...
 
+   Secure domain (SecDom)
+      A dedicated domain which executes a pre-compiled firmware component that is signed by Nordic Semiconductor.
+      It exposes security services to the other cores through an IPC interface.
+
+   Serial Peripheral Interface (SPI)
+      An interface bus commonly used to send data between microcontrollers and small peripherals such as shift registers, sensors, and SD cards.
+
+   Serial Peripheral Interface Master (SPIM)
+      A peripheral that can communicate with multiple slaves using individual chip select signals for each of the slave devices attached to a bus.
```
[the remark this answers is not in the document]
Comment (FrancescoSer) [reply doc/nrf/glossary.rst:754]: I wondered about the same @hubertmis Is slave really the correct term here?

[the remark this answers is not in the document]
Comment (FrancescoSer) [reply doc/nrf/glossary.rst:754]: Ok, thank you!

**Review by wiba-nordic — APPROVED** [review]

File: doc/nrf/glossary.rst [hunk 4]
```diff
@@ -714,6 +811,9 @@ Glossary
    Supervisor Call (SVC)
       It generates a software exception in which access to system resources or privileged operations can be provided.
 
+   System Controller
+      A :term:`VPR core` that implements system startup and power management functionalities that in the past would have been implemented in hardware.
```
Comment (wiba-nordic) [inline doc/nrf/glossary.rst:815]:
```suggestion
      A :term:`VPR core<VPR Core (pronounced “Viper”)>` that implements system startup and power management functionalities that in the past would have been implemented in hardware.
```
My mistake on this one, got the link wrong.

**Review by wiba-nordic — APPROVED** [review]

**Comment by gmarull** [comment]
@FrancescoSer please, do not blindly merge content from -next to the upstream repo. For example, all `@soc1` references do not exist here.

**Comment by FrancescoSer** [comment]
@gmarull, please check with with @shanthanordic that docs would go in as-is and would undergo revisions towards their final form before the the next devtag. If you want additional details, feel free to contact me on teams.

**Comment by gmarull** [comment]
> @gmarull, this was not a blind merge. As agreed with @shanthanordic weeks ago, docs would go in as-is and would undergo revisions towards their final form before the the next devtag. As you can notice, the pages are currently all orphaned. Moreover, they have already diverged from their version on nrf-next. If you want additional details, feel free to contact me on teams.

Just noting as this has caused failures to other PRs, so we should not copy content as is.

**Comment by FrancescoSer** [comment]
@gmarull The content was reviewed. Could you please be more specific? If you are referring to the sample board rows, there was no entry previously present for anything related to 54H in there.

**Comment by gmarull** [comment]
> If you are referring to the sample board rows, 

yes, upstream boards have names aligned with what's printed on the board

**Comment by FrancescoSer** [comment]
Until the upstream meeting on Friday, SOC1 was expected to be supported in version 2.6. Anyway, I'm glad this issue was identified.

**Pull request merged**